### PR TITLE
Adds native metadata parsing for base metadata

### PR DESCRIPTION
**Pull request by u9g**
adds a breaking change to the library

### CODE REVIEW

**Comment by u9g** [comment]
Sources: 
- old data: https://wiki.vg/index.php?title=Entity_metadata&oldid=7415#Entity
- new data: https://wiki.vg/Entity_metadata#Entity_Metadata_Format

**Comment by extremeheat** [comment]
This is something that should go into the protocol definition, but I think the version loader is a good idea

**Comment by u9g** [comment]
starts to address https://github.com/PrismarineJS/mineflayer/issues/1263

File: index.js [hunk 1]
```diff
@@ -21,6 +28,19 @@ function Entity (id) {
   this.heldItem = this.equipment[0] // shortcut to equipment[0]
   this.isValid = true
   this.metadata = []
+  // all mc versions supported by mineflayer have:
+  this.isOnFire = () => Boolean(this.metadata[0] & 0x01)
```
Comment (extremeheat) [inline index.js:32]: to be complete, can you add setting too?

**Review by extremeheat — COMMENTED** [review]
Ah, I see what's going on. It looks way more complicated in the protocol than it needs to be, so yeah I think it makes sense to do it here.

File: index.js (hunk 1 above)
Comment (extremeheat) [reply index.js:32]: would be a good idea to turn this to a class

**Review by rom1504 — CHANGES_REQUESTED** [review]
How did you check this works for most versions ? Any idea how to automatically test it ?

**Comment by u9g** [comment]
The reason this is being slowed down isnt because we didnt have a way to test it, i was going to get the info from burger, just havent got around to usung burger once yet

**Comment by u9g** [comment]
should be done using data from mcData

**Pull request closed without merging**